src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor_Minimal.cs

```diff
@@ -196,26 +196,39 @@ private void MinimallyQualify(INamedTypeSymbol symbol)
             // NOTE(cyrusn): If we're currently in a block of usings, then we want to collect the
             // aliases that are higher up than this block.  Using aliases declared in a block of
             // usings are not usable from within that same block.
-            var usingDirective = GetAncestorOrThis<UsingDirectiveSyntax>(startNode);
-            if (usingDirective != null)
+            while (startNode != null)
```

Comment:
>while (startNode != null) [](http://example.com/codeflow?start=12&length=25)

It looks like, if we are not in in a using directive, we will traverse syntax all the way to the top in this loop, and then restart reversion from the beginning looking for a namespace declaration or a compilation unit node. Instead, we could also stop once we reached a ```BaseNamespaceDeclarationSyntax``` or  ```CompilationUnitSyntax``` node. Then the next loop could continue from the current ```startNode``` instead of resetting it.

Reply:
I think I'd prefer to pass on this unless you feel strongly. It seems like the current code is authentic to the original implementation, and making the change that you suggested will complicate things slightly. If this method shows up in CPU profiles, then this suggestion sounds like a good idea to pursue.